core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -333,7 +333,7 @@ object TopicCommand extends Logging {
         }
 
         if(tp.hasPartitions) {
-          if (Topic.INTERNAL_TOPICS.contains(topic)) {
+          if (Topic.isInternal(topic)) {
             throw new IllegalArgumentException(s"The number of partitions for the internal topics${Topic.INTERNAL_TOPICS} cannot be changed.")
```

Comment:
We may as well change the log message also. I don't see why it doesn't mention `topic` directly.

Reply:
Done. Updated log comment. Also changed visibility of INTERNAL_TOPICS constant so that in future clients only use Topic::isInternal method to check if a topic is internal.